zebra/zebra_rib.c

```diff
@@ -3284,22 +3255,24 @@ int rib_sweep_kernel_reconcile_route(struct thread *thread)
 	zlog_notice("kernel_reconcile: timer_expire before sweep: add %lu, del %lu",
 		zrouter.zebra_stale_rt_add, zrouter.zebra_stale_rt_del);
 
-		if (!kernel_reconcile)
-			return;
+	if (!kernel_reconcile)
+		return -1;
 
-		RB_FOREACH (vrf, vrf_id_head, &vrfs_by_id) {
-			if ((zvrf = vrf->info) == NULL)
-				continue;
+	RB_FOREACH (vrf, vrf_id_head, &vrfs_by_id) {
```

Comment:
One thing note here is that, with iterating over all VRFs and iterating over all routes with in each VRF forms a tight loop here. There are no yields present either in old code or in the newly added code. This will be particularly a concern in scale scenarios when number of routes present in Zebra goes beyond 100,000.

Reply:
@srimohans 
Yeah, Valid point. 
I have one doubt. Right now all Zebra threads, which update RIB, runs to completion. Is this assumption correct.
If yes, how is it achieved?  By acquiring single LOCK by all zebra thread?

If no, then I need to understand more about it before adding yield here. Let me know if this is absolute necessary right now, because as you mentioned old code had this problem. Thanks.

Reply:
Since there is a discussion going on about yielding during loop or not. I will improve it by adding a check as below:
-----
		/*
		 * Avoid scanning tables as soon as all Stale routes are cleaned.
		 * If no stale route was cleaned before calling this functions
		 * first time, then we may not be learning any route, so a loop
		 * here is fine.
		 */
		if (zrouter.zebra_stale_rt_add == zrouter.zebra_stale_rt_del)
			break;
----
Thanks.

Reply:
Since we are planning to place a yield in rib_sweep_table part after client->zebra GR analysis, I have not done it with this PR. 
Improving code to make sure that sweep function scans no further, if no kernel stale rt remains. Thanks. 
Kindly review if any other changes are needed.